### PR TITLE
Configure the folder that new Finder windows open

### PR DESCRIPTION
**Pull request by aschleck**

### CODE REVIEW

**Review by Enzime — CHANGES_REQUESTED** [review]
Can you add tests?

**Comment by aschleck** [comment]
Sure, done

File: modules/system/defaults/finder.nix [hunk 1]
```diff
@@ -96,5 +96,13 @@ with lib;
       '';
     };
 
+    system.defaults.finder.NewWindowTarget = mkOption {
+      type = types.nullOr types.str;
```
Comment (Enzime) [inline modules/system/defaults/finder.nix:100]:
```suggestion
      type = types.nullOr types.enum [ "PfHm" ... ];
```

If I understand correctly, only these values are possible:

https://gist.github.com/ChristopherA/98628f8cd00c94f11ee6035d53b0d3c6#file-macos-preferences-defaults-sh-L280-L286

File: modules/system/defaults/finder.nix [hunk 2]
```diff
@@ -96,5 +96,13 @@ with lib;
...
+      default = null;
+      description = ''
+        Change the default folder shown in Finder windows. Use "PfHm" to default to the home folder.
+        The default is unset ("Recents").
+      '';
+    };
   };
```
Comment (Enzime) [inline modules/system/defaults/finder.nix:107]: Could you also add `NewWindowTargetPath`? I'm assuming that's the `defaults` option when setting `NewWindowTarget` to `Other`

**Review by Enzime — CHANGES_REQUESTED** [review]

File: modules/system/defaults/finder.nix (hunk 1 above)
Comment (aschleck) [reply modules/system/defaults/finder.nix:127]: Good call. I also added PfID which opens iCloud Drive

File: modules/system/defaults/finder.nix (hunk 2 above)
Comment (aschleck) [reply modules/system/defaults/finder.nix:136]: Done

File: modules/system/defaults/finder.nix [hunk 3]
```diff
@@ -96,5 +96,42 @@ with lib;
...
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Sets the URI to open when NewWindowTarget is "PfLo". Spaces and similar characters must be
+        escaped.
+        Example: "file:///Users/foo/cats%20and%20dogs".
+        The default is the home folder.
+      '';
+    };
   };
```
[the remark this answers is not in the document]
Comment (aschleck) [reply modules/system/defaults/finder.nix:140]: Thank you for making the suggested change here! Those make sense to me

File: modules/system/defaults/finder.nix [hunk 4]
```diff
@@ -96,5 +96,42 @@ with lib;
...
+      type = types.nullOr (types.enum [
+        "PfAF"
+        "PfCm"
+        "PfDe"
+        "PfDo"
+        "PfHm"
+        "PfID"
+        "PfLo"
+        "PfVo"
+      ]);
```
[the remark this answers is not in the document]
Comment (aschleck) [reply modules/system/defaults/finder.nix]: Had to explicitly check that key is non-null but this sounds good, thanks for suggesting

File: modules/system/defaults/finder.nix [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 with lib;
```
Comment (Enzime) [inline modules/system/defaults/finder.nix:3]:
```suggestion
```

File: modules/system/defaults/finder.nix [hunk 6]
```diff
@@ -2,6 +2,9 @@
...
 
+let
+  cfg = config.system.defaults.finder;
+in
```
Comment (Enzime) [inline modules/system/defaults/finder.nix:7]:
```suggestion
let
  inherit (lib) mkOption types ...;

  cfg = config.system.defaults.finder;
in
```

**Review by Enzime — CHANGES_REQUESTED** [review]

File: modules/system/defaults/finder.nix [hunk 7]
```diff
@@ -96,5 +99,54 @@ with lib;
...
+  };
+
+  config = {
+    assertions = [{
+      assertion = cfg.NewWindowTargetPath != null -> cfg.NewWindowTarget == "PfLo";
+      message = "`system.defaults.finder.NewWindowTarget` should be set to `Other` when `NewWindowTargetPath` is non-null.";
+    }
+    {
+      assertion = cfg.NewWindowTarget == "PfLo" -> cfg.NewWindowTargetPath != null;
+      message = "`system.defaults.finder.NewWindowTarget` should be non-null when `NewWindowTarget` is set to `Other`.";
```
Comment (dpomykala) [inline modules/system/defaults/finder.nix:149]: Hey, I noticed a small typo in the message for the second assertion. I think it should be `NewWindowTargetPath` in the first part of the message (currently `NewWindowTarget` is mentioned twice).

```suggestion
      message = "`system.defaults.finder.NewWindowTargetPath` should be non-null when `NewWindowTarget` is set to `Other`.";
```

BTW. Great job with this feature. I'm looking forward to this PR!

File: modules/system/defaults/finder.nix (hunk 6 above)
Comment (aschleck) [reply modules/system/defaults/finder.nix]: Woah this is neat syntax, thanks for suggesting it

**Review by Enzime — APPROVED** [review]
LGTM

Thanks for the contribution

**Pull request merged**